core/src/ast/types.rs

```diff
@@ -720,6 +721,16 @@ impl TypeName {
         }
     }
 
+    pub fn callback_ret_to_syn(&self) -> syn::Type {
```

Comment:
issue: I think this should be a part of `ffi_safe_version()`

(unless I'm missing something)

Reply:
That will also correctly handle `Result<Option<Foo>>` and friends